GVFS/GVFS.Virtualization/FileSystemCallbacks.cs

```diff
@@ -678,14 +700,20 @@ private FileSystemTaskResult ExecuteBackgroundOperation(FileSystemTask gitUpdate
                                         foreach (DirectoryItemInfo itemInfo in this.context.FileSystem.ItemsInDirectory(Path.Combine(this.context.Enlistment.WorkingDirectoryRoot, folderPath)))
                                         {
                                             string itemVirtualPath = Path.Combine(folderPath, itemInfo.Name);
-                                            if (itemInfo.IsDirectory)
+                                            string oldItemVirtualPath = gitUpdate.OldVirtualPath + itemVirtualPath.Substring(gitUpdate.VirtualPath.Length);
+                                            if (!this.newlyCreatedFileAndFolderPaths.Contains(itemVirtualPath))
                                             {
-                                                relativeFolderPaths.Enqueue(itemVirtualPath);
+                                                this.newlyCreatedFileAndFolderPaths.Add(itemVirtualPath);

```

Comment:
We are adding the top level folder `result = this.TryAddModifiedPath(gitUpdate.VirtualPath, isFolder: true);` so we don't need to add all the subfolder and files.  This is kind of left over from when we would add the top level folder to that sparse-checkout and then need to add files to the always_exclude.  Since this is just one list now we only need the top level folder and git will handle it correctly for both the sparse and exclude cases.